### PR TITLE
Example terraform scripts for installing MKE on GCP

### PR DESCRIPTION
**Pull request by ranyodh**
## Description
This is an example of terraform scripts that prepares a cluster
on Google Cloud Platform to install Mirantis Kubernetes Engine.

It creates following GCP resources:
 - A VPC with specified MTU (default value is 1500) and one subnetwork
 - Linux and Windows instances using public images provided by GCP
 - Firewall rules to allow internal and external traffic
 - A network load balancers for MKE and MSR that targets corresponding  unmanaged instance group
(along with health checks)

### MTU:
The default MTU is set to 1500 (rather than the default 1460). This is to an issue where swarm services
fails because Docker network uses 1500 as the default MTU.

## Known Issues:
* MSR (Mirantis Secure Registry) installation fails due to invalid SSL certificate for the load balancer IP
* Windows nodes does not work when it is assigned "Kubernetes" orchestration

### CODE REVIEW

**Review by james-nesbitt — APPROVED** [review]
This looks good to me.

I would have preferred some more sane defaults, but your pattern matches what is in place for AWS.

I don't have credentials to test it, but I think that it is a good start.

**Comment by ranyodh** [comment]
> This looks good to me.
> 
> I would have preferred some more sane defaults, but your pattern matches what is in place for AWS.
> 
> I don't have credentials to test it, but I think that it is a good start.

Yes, I based the instance types, disk sizes etc on AWS examples.

File: examples/terraform/azure/modules/master/main.tf [hunk 1]
```diff
@@ -317,6 +317,7 @@ EOF
     ssh_keys {
       path     = "/home/ubuntu/.ssh/authorized_keys"
       key_data = var.ssh_key.public_key_openssh
+
```
Comment (sakshisharma84) [inline examples/terraform/azure/modules/master/main.tf]: Can we delete this extra line?

File: examples/terraform/gcp/main.tf [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        address  = host.network_interface.0.access_config.0.nat_ip
+        user     = var.windows_user
+        password = var.windows_password
+        useHTTPS = true
+        insecure = true
+      }
+      role             = host.metadata["role"]
+      privateInterface = "Ethernet"
+    }
+  ]
```
Comment (sakshisharma84) [inline examples/terraform/gcp/main.tf:142]: We can add a newline for better readability.

**Review by sakshisharma84 — APPROVED** [review]
LGTM! Just minor comments

**Review by kke — APPROVED** [review]

File: examples/terraform/gcp/main.tf (hunk 2 above)
Comment (ranyodh) [reply examples/terraform/gcp/main.tf:142]: done

File: examples/terraform/azure/modules/master/main.tf (hunk 1 above)
Comment (ranyodh) [reply examples/terraform/azure/modules/master/main.tf]: done

**Pull request merged**